Common/SemanticAnalyzer/src/AstStatementParser.cpp

```diff
@@ -0,0 +1,185 @@
+#include "AstStatementParser.h"
+
+
+std::vector<std::pair<std::string, std::any>>& SEMANTICANALYZER::AstStatementParser::parse()
+{
+    for(auto it = astTree.begin(); it != astTree.end(); ++it)
+    {
+        std::pair<std::string, std::any> statement;
+
+        if(it->value == OBJECT_DECL)
+        {
+            statement.first = OBJECT_DECL;
+            serializeObjectDec(it, std::any_cast<ObjectDecl&>(statement.second));
+        }
+        else if(it->value == RELATION)
+        {
+            statement.first = RELATION;
+            serializeRelation(it, std::any_cast<Relation&>(statement.second));
+        }
+        else if(it->value == NOTE)
+        {
+            statement.first = NOTE;
+            serializeNote(it, std::any_cast<Note&>(statement.second));
+        }
+        else if(it->value == GRAPH)
+        {
+            statement.first = GRAPH;
+            serializeGraph(it, std::any_cast<Graph&>(statement.second));
+        }
+        else if(it->value == DOT_CLOUD)
+        {
+            statement.first = DOT_CLOUD;
+            serializeDotCloud(it, std::any_cast<DotCloud&>(statement.second));
+        }
+
+        programTree.push_back(statement);
+    }
+
+    return programTree;
+}
+
+
+void SEMANTICANALYZER::AstStatementParser::serializeObjectDec(AST::ASTTree::DSFIterator& objectDecIter, ObjectDecl& objectDecl)
+{
+    for(auto it = objectDecIter; it != AST::ASTTree::DSFIterator(); ++it)
```

Comment:
```suggestion
    for(auto it = objectDecIter; it != astTree.end(); ++it)
```
[minor] Я бы предложил лучше так, потому что это и есть end, а то так сходу не понятно, до куда идти (ниже аналогично)

Reply:
Спасибо, я благодаря этому нашел важную ошибку в коде. Но не совсем согласен с замечанием, astTree.end() - это конец целого дерева, я же имею итератор на какую-то его часть, и поэтому когда цикл заканчивает, я делаю обход лишь некоторого поддерева для конкретного statement.

Reply:
Да, понял, тогда у тебя всё правильно. Этот итератор кривоват, поэтому только так. Единственное, что хочу предупредить, что так как ты обошёл поддерево своим новым итераттором, то основной итератор всё равно обойдёт это поддерево, но так как у тебя в основном цикле проверки на statement, то всё сработает